pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -132,17 +126,12 @@ def advance(self, batch, batch_idx):
         self.trainer.logger_connector.on_train_split_start(batch_idx, split_idx, split_batch)
 
         if self.trainer.lightning_module.automatic_optimization:
-            for opt_idx, optimizer in self.get_active_optimizers(batch_idx):
-                # handle optimization restart
-                if self.restarting:
-                    if opt_idx < self.optim_progress.optimizer_idx:
-                        continue
-
-                self.optim_progress.optimizer_idx = opt_idx
-
-                result = self._run_optimization(batch_idx, split_batch, opt_idx, optimizer)
-                if result:
-                    self.batch_outputs[opt_idx].append(copy(result.result_collection))
+            # in automatic optimization, hand over execution to the OptimizerLoop
+            optimizers = [optimizer for _, optimizer in self.get_active_optimizers(batch_idx)]
```

Comment:
Should the `OptimizerLoop` takes care of active `optimizers` creation and this would be moved to `on_run_start` and used for self.done ?

Reply:
for the other loops, we are also creating the iterable outside and pass it to run()
e.g. in fit loop where we create the dataloader and then pass it to the epoch loop. 

I was following the same pattern here.

Reply:
I would prefer to have the splitting of the batch inside the loop that corresponds to the TBPTT, which is TrainingBatchLoop. This way the loop can be customized and the parent loop does not need knowledge about whether the child loop is requiring a split batch or not.